### ISSUE TITLE
Allow editing certain repository settings

### ISSUE THREAD

**Issue by SStorm** [crate/crate#14148]
### Problem Statement

At the moment, it's possible to create repositories, but not update them in any way. It would be useful to have some of the settings on repositories editable. For example, for s3 repos, it would be great to be able to edit the access and secret keys, because best practices says they must be rotated from time to time. 

Also the clauses in https://crate.io/docs/crate/reference/en/5.3/sql/statements/create-repository.html#clauses, because often times we want to increase `max_restore_bytes_per_sec` when restoring large snapshots.

**Comment by mfussenegger** [comment]
As a workaround it's possible to drop the repository and re-create it with the new settings.

It doesn't delete the snapshots. Downside is that you need to know all the existing settings and that the repository information will be gone for the short period.

**Comment by SStorm** [comment]
Yes the workaround is ok, but as you said - it's quite inconvenient.

**Comment by hammerhead** [comment]
Another case is the increase of `max_snapshot_bytes_per_sec`. In one scenario, we had a cloud cluster with significant data volume that was upgraded to a new minor CrateDB version. The upgrade triggers a rotation of repositories and therefore a full new snapshot.
While the snapshot was still running, and limited to the default of 40 MB/s, an upgrade-related issue was discovered that needed a full cluster restart to address. The initial snapshot was estimated to take 14 hours. With the ability to alter `max_snapshot_bytes_per_sec` the snapshot creation could have been accelerated to finish quickly enough to not delay the full restart too much.

**Comment by seut** [comment]
The team decided to implement this in the first iteration without supporting live updates to ongoing restores/snapshots, current estimation of 1-3 reflects this.
